internal/statesync/stateprovider.go

```diff
@@ -411,16 +411,12 @@ func (s *stateProviderP2P) tryGetConsensusParamsFromWitnesses(
 				Height: uint64(height),
 			},
 		}); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("problem sending request: %w", err)
 		}
-
-		timer.Reset(consensusParamsResponseTimeout)
 		select {
 		// if we get no response from this provider we move on to the next one
-		case <-timer.C:
-			continue
 		case <-ctx.Done():
-			return nil, ctx.Err()
+			return nil, fmt.Errorf("timed out waiting for response for consensus params request: %w", ctx.Err())
```

Comment:
But this logic will never loop through the witnesses. It will just hang on the first witness until the context times out. Or am I missing something?

Reply:
I was incorrect about the behavior of taking the second value out of a channel. 

The weird part is that this causes the test to pass locally (where they were failing quickly for me locally.

Reply:
In any case, I rewrote the retry mechanism in a way that is consistent with the (your) original intent, but that is kind of _a lot_.